src/Source/WebformSubmissionCount.php

```diff
@@ -24,28 +24,72 @@ public static function forField($entity, $field, $instance) {
   public function __construct($entity) {
     $this->entity = $entity;
   }
+
   /**
    * Get the value for the given item.
    *
    * @return int
-   *   The number of webform submissions in $this-entity
-   *   and all it's translations.
+   *   The number of webform submissions in $this-entity,
+   *   and on additional nodes provided via the field widget,
+   *   and all their translations.
    */
   public function getValue($item) {
     $entity = $this->entity;
-    $q = db_select('node', 'n');
+    $nids = array($entity->nid);
+    if (isset($item['options']['source']['add_nids'])
+        && $add_nids = $item['options']['source']['add_nids']) {
+      array_push($nids, ...$add_nids);
+    }
+    $q_nids = db_select('node', 'n');
+    $q_nids->leftJoin('node', 'nt', 'n.tnid>0 AND nt.tnid=n.tnid');
+    $q_nids->addExpression('COALESCE(nt.nid, n.nid)', 'nid');
+    $q_nids->condition('n.nid', $nids);
+
+    $q = db_select('webform_submissions', 'ws');
     $q->addExpression('COUNT(ws.nid)');
-    $q->innerJoin('webform_submissions', 'ws', 'n.nid=ws.nid');
-    $q->where(
-      'n.nid=:nid OR ((n.nid=:tnid OR n.tnid=:tnid) AND :tnid>0)',
-      array(':nid' => $entity->nid, ':tnid' => $entity->tnid)
-    );
+    $q->condition('ws.nid', $q_nids, 'IN');
     return $q->execute()->fetchField();
   }
+
   /**
-   * No extra configuration for the widget needed.
+   * Build the configuration form for the field widget.
+   *
+   * Add a field to include submissions from other nodes in the count.
    */
   public function widgetForm($item) {
-    return NULL;
+    $source_options = isset($item['options']['source']) ? $item['options']['source'] : array();
+    $source_options += array('add_nids' => '');
+    if (is_array($source_options['add_nids'])) {
+      $source_options['add_nids'] = implode(',', $source_options['add_nids']);
+    }
+    $form['add_nids'] = array(
+      '#title' => t('Include nodes'),
+      '#description' => t('Enter node ids (separated by comma) to add submissions from those nodes (and their translations) to the number shown by the counter.'),
+      '#type' => 'textfield',
+      '#default_value' => $source_options['add_nids'],
+    );
+    $form['#element_validate'][] = array($this, 'widgetValidate');
+    return $form;
+  }
+
+  /**
+   * Element validate callback for @see widgetForm().
+   */
+  function widgetValidate($element, &$form_state, $form) {
```

Comment:
A lot of this functionality seems to be duplicated in both plugins. We might want to move the functionality to it’s own class and use that in both of the plugins (composition or inheritance).